web/modules/custom/joinup_comment/joinup_comment.module

```diff
@@ -23,8 +24,17 @@ function joinup_comment_comment_view_alter(array &$build, EntityInterface $entit
 /**
  * Implements hook_form_FORM_ID_alter().
  *
- * Change the submit button label for the comment form.
+ * Disallow commenting for anonymous visitors and
+ * change the submit button label for the comment form.
  */
-function joinup_comment_form_comment_form_alter(&$form, FormStateInterface $form_state, $form_id) {
+function joinup_comment_form_comment_form_alter(array &$form, FormStateInterface $form_state, string $form_id) {
+  if (Drupal::currentUser()->isAnonymous()) {
+    $form['#access'] = FALSE;
+    $login_link = Link::createFromRoute('Login', 'user.login')->toString();
+    $register_link = Link::createFromRoute('create an account', 'user.register')->toString();
+    $text = sprintf('%s or %s to comment.', $login_link, $register_link);
+    $form['#markup'] = '<div class="center">' . $text . '</div>';
```

Comment:
The div wrappers should be part of the `#prefix` and `#suffix` elements of the form, not as inline markup.